New Module: Add bitbucket_deploy_key (source_control/bitbucket_deploy_key)

bincyber opened the pull request:
##### SUMMARY

Added a new module for managing deploy keys for BitBucket repositories.

Currently, this can be accomplished using the [uri](https://docs.ansible.com/ansible/latest/uri_module.html) module, however it requires some effort to achieve idempotency. It's better to have a specific module that can do the same thing in 1 task.

This is similar to the github_deploy_key module. See ansible/ansible#24364

##### ISSUE TYPE
 - New Module Pull Request


##### COMPONENT NAME

bitbucket_deploy_key


##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'modules/']
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```


##### ADDITIONAL INFORMATION

bincyber committed:
Added new module: bitbucket_deploy_key

- added new module to manage deploy keys for BitBucket repositories

bincyber commented:
I don't know what is causing the CI tests to fail? Can someone assist me with getting the tests to pass?

rickdgeerling commented:
It seems like the Ansible team has set up some [custom linting rules](https://github.com/ansible/community/wiki/Testing:-boilerplate,-wildcard-imports,-and-get_exception). (Found by following a link in `test/sanity/code-smell/boilerplate.sh`)

When I look at the existing [`github-deploy-keys`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/github_deploy_key.py#L6) integration, it seems like the team wants these two lines in every file.

```
from __future__ import absolute_import, division, print_function
__metaclass__ = type
```

Just my 2ct, hope that't it because I would like to use this module :)

bincyber committed:
Modified module to pass CI tests

bincyber commented:
Thank you for your help @tuxmachine. Adding those two lines did the trick.

terbolous commented:
This has been stale for a while. Are there anything outstanding works preventing this from being reviewed/merged?

eriweb commented on lib/ansible/modules/source_control/bitbucket_deploy_key.py:
```diff
@@ -0,0 +1,271 @@
+            key_id = response_body["pk"]
+            self.module.exit_json(changed=True, msg="Deploy key successfully added", pk=key_id)
+        elif status_code == 400:
+            existing_key_msg = "Someone has already registered this key as a deploy key for this repository"
```
Based on my testing it seems bitbucket has changed the response.
This seems to work:
`existing_key_msg = "Someone has already added that access key to this repository."`

bincyber replied:
Thank you for trying this module. I've updated the code to reflect these changes.

bincyber committed:
Updated bitbucket_deploy_key module

- updated message returned when there is an existing deploy key
- bumped version_added to 2.8
- fixed issues reported by ansible-test

bincyber commented:
ready_for_review

eriweb reviewed (approved):
Used this module for a couple of days, so far no issues

Akasurde commented:
@bincyber Thanks for the contribution. @terbolous Thanks for the review.
